confgenerator/logging_processors.go

```diff
@@ -39,77 +44,182 @@ func init() {
 	LoggingProcessorTypes.RegisterType(func() Component { return &LoggingProcessorParseJson{} })
 }
 
-// A LoggingProcessorParseRegex applies a regex to the specified field, storing the named capture groups as keys in the log record.
-type LoggingProcessorParseRegex struct {
-	ConfigComponent             `yaml:",inline"`
-	LoggingProcessorParseShared `yaml:",inline"`
+// A LoggingProcessorParseRegexSimple applies a regex to the specified field, storing the named capture groups as keys in the log record.
+// This was maintained in addition to the parse_regex_complex to ensure backward compatibility with any existing configurations
+type LoggingProcessorParseRegexSimple struct {
+	ConfigComponent        `yaml:",inline"`
+	fluentbit.ParserShared `yaml:",inline"`
+	Field                  string `yaml:"field,omitempty"`
 
 	Regex string `yaml:"regex,omitempty" validate:"required"`
 }
 
-func (r LoggingProcessorParseRegex) Type() string {
+func (r LoggingProcessorParseRegexSimple) Type() string {
 	return "parse_regex"
 }
 
-func (p LoggingProcessorParseRegex) Components(tag, uid string) []fluentbit.Component {
-	filter, parser := p.LoggingProcessorParseShared.Components(tag, uid)
+func (p LoggingProcessorParseRegexSimple) Components(tag, uid string) []fluentbit.Component {
+	parser, parserName := p.ParserShared.Component(tag, uid)
 	parser.Config["Format"] = "regex"
 	parser.Config["Regex"] = p.Regex
+
 	return []fluentbit.Component{
-		filter,
 		parser,
+		fluentbit.ParserFilterComponent(tag, p.Field, []string{parserName}),
+	}
+}
+
+type RegexParser struct {
+	Regex  string                 `yaml:"regex,omitempty" validate:"required"`
+	Parser fluentbit.ParserShared `yaml:",inline"`
+}
+
+// A LoggingProcessorParseRegex applies a set of regexes to the specified field, storing the named capture groups as keys in the log record.
```

Comment:
Since we're not exposing any of these processors to users, can we simplify these structs? It seems like there's a lot of redundancy here (e.g. `Type` methods, yaml struct tags, etc.) that isn't being used at all.